### PR TITLE
Spec for PATCH /v4/organizations/:org-id/

### PR DESCRIPTION
**Pull request by oponder**
Discussed here: https://github.com/giantswarm/api-spec/issues/29

I think there are a few more things to discuss though:

**1. What to do when the request is invalid?**
Currently I have specced it so that the entire patch operation fails. So if a member is being added that doesn't exist, or has an invalid email address, no  add/remove of members will happen at all, even if there might be other members in the patch request that are valid. Do we agree this is the right way to go about it? An alternative would be to just ignore invalid input.

**2. I think we need a new giant swarm response code:**
We have `UNKNOWN_ATTRIBUTE` and `IMMUTABLE_ATTRIBUTE` but that doesn't really match with what is going wrong here. What do you think of `INVALID_INPUT` with a http error code of `400` ?

ping @marians

### CODE REVIEW

**Comment by marians** [comment]
1. When part of the request is invalid (non-existing user), the entire request should fail and no changes should be applied. This way it's much easier for a client to understand what's been done and what hasn't.

2. `INVALID_INPUT` with 400 is fine with me. If you like, we could even be more specific for this case by using a code like `INVALID_USER`.

It would be nice to sort the responses in the spec.yaml by response code, ascending (default at the end).

**Comment by oponder** [comment]
I think `INVALID_INPUT` should cover it for now, otherwise we'll be probably be adding `INVALID_CLUSTER`, `INVALID_KEYPAIR` and `INVALID_ORGANIZATION` too.

File: spec.yaml [hunk 1]
```diff
@@ -446,6 +446,79 @@ paths:
...
+        - `members`: By modifying the array of members, members can be added or removed from the organization
+
+        The request body must conform with the [JSON Patch Merge (RFC 7386)](https://tools.ietf.org/html/rfc7386) standard.
+        Requests have to be sent with the `Content-Type: application/merge-patch+json` header.
+
+        The full request must be valid before it will be executed, currently this
+        means every member you attempt to add to the organization must actually
+        exist in the system. If any member you attempt to add is invalid, the entire
+        patch operation will fail, no members will be added or removed, and an error message
+        will explaining which members in your request are invalid.
```
Comment (marians) [inline spec.yaml]: explaining -> explain

File: spec.yaml [hunk 2]
```diff
@@ -446,6 +446,79 @@ paths:
...
+                  $ref: 'definitions.yaml#/definitions/V4OrganizationMember'
+
+      summary: Modify organization
+      description: |
+        This operation allows you to modify an existing organization. You must be
+        a member of the organization or an admin in order to use this endpoint.
+
+        The following attributes can be modified:
+
+        - `members`: By modifying the array of members, members can be added or removed from the organization
```
Comment (marians) [inline spec.yaml]: or "added to or removed from" ?

**Comment by oponder** [comment]
Thx for reviewing!

**Pull request merged**